lib/pbio/src/util.c

```diff
@@ -3,6 +3,58 @@
 
 #include <stdbool.h>
 #include <stdint.h>
+#include <string.h>
+
+/**
+ * Compares a 128-bit UUID from @p le in little endian format with
+ * the given uuid @p uuid.
+ *
+ * According to RFC 4122, the UUID is grouped into the following:
+ * 1) One 32-bit
+ * 2) Two 16-bit
+ * 3) Eight 8-bit
+ *
+ * @param [in]  le      The little endian UUID.
+ * @param [in]  uuid    The UUID to compare against.
+ */
+bool pbio_uuid128_le_compare(const uint8_t *le, const uint8_t *uuid) {
+    return (le[0] == uuid[3]) &&
+           (le[1] == uuid[2]) &&
+           (le[2] == uuid[1]) &&
+           (le[3] == uuid[0]) &&
+           (le[4] == uuid[5]) &&
+           (le[5] == uuid[4]) &&
+           (le[6] == uuid[7]) &&
+           (le[7] == uuid[6]) &&
+           (memcmp(&le[8], &uuid[8], 8) == 0);
+}
+
+/**
+ * Copies a 128-bit UUID from @p src to a buffer @p dst,
+ * which is a buffer used by a little endian medium.
+ *
+ * According to RFC 4122, the UUID is grouped into the following:
+ * 1) One 32-bit
+ * 2) Two 16-bit
+ * 3) Eight 8-bit
+ *
+ * @param [in]  dst     The destination array.
```

Comment:
```suggestion
 * @param [in]  dst     The destination that will receive the resulting RFC 4122 formatted UUID.
```

Reply:
RFC 4122 is not a format for a UUID, it just explains that parts of the 128-bit UUID are grouped into 16- and 32-bit values, so encoding those on a little endian medium (like USB) involves some byte-swapping. I would suggest we don't modify this comment, or change it to something else.

Reply:
Sure, I think saying that this is the format required for USB is the most useful. The main point of this suggestion though was that it wasn't entirely clear to me which parameter uses which format.